mautic/mautic
# Fix for Amazon API transport return-path when sending emails immediately

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 4.x
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
Email messages sent immediately through Amazon SES API do not include the return-path property.  Custom return path (bounce) address works fine when email is sent through queue.

This PR also includes minor `AmazonApiTransport` and `AmazonApiTransportTest` code cleanup (types declaration, comment fixes, unnecessary statements removed, etc.).



#### Steps to test this PR:
1. Setup Mautic with `Amazon SES - API` email setting
2. Set `Custom return path (bounce) address`
3. Set `How should email be handled?` to `Send immediately`
4. Send email to contact that will bounce, using `From email` other than `Custom return path (bounce) address`
5. The Delivery Status Notification should be sent to `Custom return path (bounce) address`, not `From email`




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10474"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
